filebeat/check.py

```diff
@@ -0,0 +1,59 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+
+# 3rd party
+import json
+import os
+
+# project
+from checks import AgentCheck
+
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'filebeat'
+
+
+class FilebeatCheck(AgentCheck):
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+
+    def check(self, instance):
+        registry_file_path = instance.get('registry_file_path')
+        if registry_file_path is None:
+            raise Exception('An absolute path to a filebeat registry path must be specified')
+
+        registry_contents = self._parse_registry_file(registry_file_path)
+
+        for item in registry_contents.itervalues():
+            self._process_registry_item(item)
+
+    def _parse_registry_file(self, registry_file_path):
+        try:
+            with open(registry_file_path) as registry_file:
```

Comment:
Indeed it is. Typically, filebeat won't track more than a handful of files per box